### ISSUE TITLE
ArgParser in extrautils and others

### ISSUE THREAD

**Issue by pi-pi3** [redox-os/redox#923]
From what I've seen none of the [extrautils](https://github.com/redox-os/extrautils) are using the [`ArgParser`](https://github.com/redox-os/coreutils/blob/master/src/lib.rs#L88) that is present within [coreutils](https://github.com/redox-os/coreutils). For unification, clarification and extensibility purposes I think it might be useful to rewrite parts of extrautils to implement ArgParser.
For that the entire `ArgParser` might need to be moved to [libextra](https://github.com/redox-os/libextra) to avoid making coreutils a dependency of extrautils. Most, if not all, core- and extrautils are already using libextra, so it wouldn't be a problem.
Additionally, coreutils' [lib.rs](https://github.com/redox-os/coreutils/blob/master/src/lib.rs) implements some other useful functions, like `format_system_time`, `format_time` and `to_human_readable_string`, which would also be useful in extrautils and other repositories.

tl;dr: `ArgParser`, `format_system_time`, `format_time` and `to_human_readable_string` should be located in libextra instead of coreutils, so they can be used inside other repositories, like extrautils. Also, extrautils should use those functionalities.

If this deems to be neccessary or useful, I'm willing to implement it.

**Comment by jackpot51** [comment]
ArgParser is now in https://github.com/redox-os/arg-parser
